### PR TITLE
Added tactical options to the small arrival vaults

### PR DESCRIPTION
**Pull request by theJollySin**
Recently, a Reddit user posted that one of the small arrival vaults was flawed in their game:

https://old.reddit.com/r/dcss/comments/fb8gi6/promising_start/

As I had *just* improved the small arrival vaults, this was disappointing to me. So I went over the small arrival vaults with a great deal more consideration, and made the smallest changes I could to improve their tactical options.

To these vaults I added an extra exit:

* evilmike_arrival_fibonacci_small
* minmay_arrival_stair_chambers
* minmay_arrival_widening_spiral_wide
* dpeg_arrival_lava_temple_mockup

And to these vaults I added a pillar:

* evilmike_arrival_grusome_pit
* minmay_arrival_widening_spiral_narrow
* minmay_arrival_loop

Removed these vaults:

* evilmike_arrival_sounds_zot
* onia_arrival_larder

I also added `no_trap_gen` to some of these if they were open to the problem the above Reddit poster faced.

### CODE REVIEW

File: crawl-ref/source/dat/des/arrival/small.des [hunk 1]
```diff
@@ -3279,7 +3288,7 @@ xxxxxxx@xxxxxxx
 ENDMAP
 
 NAME:       psy_arrival_terrarium
-TAGS:       arrival no_monster_gen
+TAGS:       arrival no_monster_gen no_trap_gen
```
Comment (ebering) [inline crawl-ref/source/dat/des/arrival/small.des]: This vault is ok with traps

File: crawl-ref/source/dat/des/arrival/small.des [hunk 2]
```diff
@@ -3308,7 +3317,7 @@ ENDMAP
 
 NAME: beargit_arrival_greenhouse
 ORIENT: float
-TAGS: arrival no_monster_gen
+TAGS: arrival no_monster_gen no_trap_gen
```
Comment (ebering) [inline crawl-ref/source/dat/des/arrival/small.des]: This vault is ok with traps

File: crawl-ref/source/dat/des/arrival/small.des [hunk 3]
```diff
@@ -110,29 +110,30 @@ ENDMAP
...
 ###############################################################################
 NAME:   evilmike_arrival_fibonacci_small
-TAGS:   arrival no_monster_gen
+TAGS:   arrival no_monster_gen no_trap_gen
 ORIENT: float
 WEIGHT: 6
 NSUBST: U = 1:+ / *:x, V = 1:+ / *:x, W = 1:+ / *:x, X = 1:+ / *:x
+NSUBST: Y = 1:+ / *:x, Z = 1:+ / *:x
 SUBST:  .= .:2500 1
 MONS:   rat
```
[the remark this answers is not in the document]
Comment (theJollySin) [reply crawl-ref/source/dat/des/arrival/small.des]: I mean, I definitely agree.  I was just trying to make the smallest possible changes.  Fixed.

File: crawl-ref/source/dat/des/arrival/small.des [hunk 4]
```diff
@@ -2156,7 +2164,7 @@ ENDMAP
 
 ##############################################################
 NAME:    onia_arrival_larder
```
[the remark this answers is not in the document]
Comment (theJollySin) [reply crawl-ref/source/dat/des/arrival/small.des]: Agreed. Removed.

File: crawl-ref/source/dat/des/arrival/small.des [hunk 5]
```diff
@@ -858,21 +861,22 @@ TAGS:    arrival no_monster_gen no_rotate
 ORIENT:  float
 SHUFFLE: cxxx
 NSUBST:  . = 1:d / *:.
+SUBST:   e = .++
```
[the remark this answers is not in the document]
Comment (theJollySin) [reply crawl-ref/source/dat/des/arrival/small.des]: I have revised my opinion. This vault needs two doors at all times.

File: crawl-ref/source/dat/des/arrival/small.des [hunk 6]
```diff
@@ -1582,23 +1588,25 @@ SHUFFLE: vXc
...
-@.{vvv''Z''vvv{.@
-..vvvv'''''vvvv..
-.vvvvv''1''vvvvv.
-vvvvvvvvvvvvvvvvv
- vvvvvvvvvvvvvvv
-  vvvvvvvvvvvvv
-   vvvvv{vvvvv
-    vvv...vvv
-     v..@..v
+      ...@...
```
[the remark this answers is not in the document]
Comment (theJollySin) [reply crawl-ref/source/dat/des/arrival/small.des]: Removed for now.

**Pull request closed without merging**